### PR TITLE
add github actions workflow yaml

### LINKED ISSUE #13: Consider building for Python 3.8
Python 3.8 has been around for a while, I think we should consider adding it to the appveyor/travis build matrices

See also DLR-SC/tixi#165.

### PR DESCRIPTION
**Pull request by joergbrech**
Ideally, this will replace appveyour and travis so that we have concurrent builds

To Do:
 - [x] make sure anaconda is setup on all builds correctly and the right python version is used
 - [x] setup anaconda api token
 - [x] delete appveyor and travis files
 - [x] change `pull_request` to `pull_request_target`
 - [x] Disable appveyor builds over at appveyor

Fixes #13

### CODE REVIEW

**Comment by rainman110** [comment]
@joergbrech Isn't there something like a miniconda action to install conda?

**Comment by rainman110** [comment]
https://github.com/marketplace/actions/setup-miniconda

**Comment by joergbrech** [comment]
I know, we are using this action for the windows tigl builds. But the default github runners only have the 64 bit Miniconda installed.

**Comment by rainman110** [comment]
32 bit win seems to be worked on:
https://github.com/goanpeca/setup-miniconda/issues/10

**Comment by rainman110** [comment]
We could for now use simply https://github.com/bryanwweber/setup-miniconda

**Comment by joergbrech** [comment]
that could work!

**Comment by joergbrech** [comment]
We need to provide the Anaconda API token to the new CI, for instance by adding it as a secret to this repo and adding

```yaml
env:
  ANACONDA_API_TOKEN: ${{ secrets.ANACONDA_API_TOKEN}}
```

to the config file. @rainman110, could you take care of this?

**Comment by joergbrech** [comment]
Apparently, Anaconda dropped 32 Bit support for Linux: https://github.com/goanpeca/setup-miniconda/pull/52#issuecomment-671349543

**Comment by joergbrech** [comment]
@rainman110, could you take over from here? There are three To-Do items left in the PR description. 
 - I don't have the permission to add secrets to this repo 
 - changing `pull_request` to `pull_request_target` should probably be the last change before merging, so we can test the workflow. 
 - I am reluctant to kill the appveyor project before we are confident that this works.

**Comment by rainman110** [comment]
I completely have forgotten this PR 🤦

**Comment by rainman110** [comment]
The first thing to do is to add secrets?

**Comment by joergbrech** [comment]
Yes I think so. But I haven't looked into this PR for a while myself 😝

**Comment by rainman110** [comment]
I added the `ANACONDA_API_TOKEN` variable to the enviroment `Anaconda`. Do we have to activate the environment somehow or is this variable used by default?

**Comment by rainman110** [comment]
@joergbrech Do you see, why the builds are failing? I don't understand this...

File: .github/workflows/build-conda-packages.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    runs-on: ${{ matrix.os }}    
+
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+
+
+    - name: Setup conda
+      uses: joergbrech/setup-miniconda@fix-10 # change this as soon as upstream PR is merged
```
Comment (joergbrech) [inline .github/workflows/build-conda-packages.yml:37]:
```suggestion
      uses: conda-incubator/setup-miniconda@v2
```

**Comment by rainman110** [comment]
Uploading from ubuntu seems to be working. Windows seems to get stuck

**Comment by rainman110** [comment]
I think we have to install the macOS SDK, as done here:  https://github.com/DLR-SC/tigl-conda/blob/4d2d5ef6cdfa2f49eb6d8f603a5bc364bea929da/.travis.yml#L47

**Comment by rainman110** [comment]
macOS compilers are fixed after downloading the 10.9 SDK.

**Comment by rainman110** [comment]
Everything looks good now. Shall be squash, or do you want to merge normally?

**Comment by joergbrech** [comment]
Nice! I can go either way, squashing is probably a bit cleaner

**Pull request merged**